src/movies.js

```diff
@@ -1,31 +1,84 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  return moviesArray.map((m) => m.director);
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  return moviesArray
+    .filter((mov) => mov.genre.includes('Drama'))
+    .filter((m) => m.director === 'Steven Spielberg').length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (!moviesArray[0]) return 0;
+  const result =
+    moviesArray
+      .filter((m) => m.score)
+      .reduce((prev, curr) => prev + curr.score, 0) / moviesArray.length;
+  return Number(result.toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  const dramaMovies = moviesArray.filter((m) => m.genre.includes('Drama'));
+  if (!dramaMovies[0]) return 0;
+  const average =
+    dramaMovies.reduce((prev, curr) => prev + curr.score, 0) /
+    dramaMovies.length;
+  return Number(average.toFixed(2));
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+  const sortedArray = [...moviesArray].sort((a, b) => {
+    if (a.year === b.year)
+      return a.title.toUpperCase() < b.title.toUpperCase()
+        ? -1
+        : a.title.toUpperCase() > b.title.toUpperCase()
+        ? 1
+        : 0;
```

Comment:
alternatively you can use `localeCompare()` , would be shorter syntax .
Ternary operator - especially if it's several conditions , can be more difficult to read ;)